### ISSUE TITLE
feature idea: start on boot option

### ISSUE THREAD

**Issue by lm1909** [jshackles/idle_master#155]
A "start on boot" option would be really great so you wouldn't always forget to start idle master and waste time (of course you can add it to your autostart manually but this would be easier and better for inexperienced users)

**Comment by Efreak** [comment]
Please don't. I'd prefer not to have things like this changing system settings. Why would you want it running all the time anyways?

**Comment by notMichael** [comment]
I understand your worries. On the other hand if you enable start on boot then it adds just one registry key which is standart way how to do it. After that you can disable start on boot any time with system's "msconfig.exe" as same as for other programs.

Why would you want it running all the time anyways?
It is small program which won't have big impact on system start and as reward you will have always as many cards as possible. This option is probably for people who have to farm like 200cards and do not want to bother start this program every time they want to see a new episode of their favourite tv show. I mean I personally won't use it because I like to run only things I need in that moment, but as a defaultly disabled feature it is not that bad.

Problem with  "not start on boot" steam
Problem is that idlemaster is dependent on running steam in the background. If someone check start on boot for idlemaster but not for steam then it simply won't work. Solution would be checking also start on boot steam and if that is disabled then user won't be able to enable start on boot for idlemaster.

**Comment by lm1909** [comment]
I fully agree with you (@notMichael) - for these users such a feature would be very useful and handy. I think it is obvious that it should be disabled by default so it won't really bother people who won't use this feature.

I think an easier solution would be to let IdleMaster just wait for steam to be open when its started and put a label with an information to enable steam on boot under the checkbox of the start on boot option, so the users have to set the start on boot option of steam themselves. This way no complicated process to check if steam is set to start on boot is enabled would be needed (but its not that nice and easy for the user as your idea).

**Comment by notMichael** [comment]
I have tried something. Where you are not able to set Start on boot if steam is not starting on boot. Then I put there a button and Tooltip information. It looks quite ugly but it looks like it is working. 

About delaying IdleMaster to start after Steam. Actualy it is not needed. I just tried it and IdleMaster check if steam is on every 500ms automaticaly so after I turned on steam everything went just fine.

**Comment by claell** [comment]
+1 for this.

**Comment by ardaozkal** [comment]
Just run Run (Win+R) and write shell:Startup, then copy a shortcut of idle master there.

Boom issue solved.

**Comment by claell** [comment]
Natively in the settings of idle master would be better :)

**Comment by ardaozkal** [comment]
https://github.com/jshackles/idle_master/pull/156

someone did it, but it needs an update as it is a few commits back.
